cmd/frontend/graphqlbackend/event_logs.go

```diff
@@ -0,0 +1,53 @@
+package graphqlbackend
+
+import (
+	"context"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/db"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/graphqlutil"
+)
+
+func (r *UserResolver) EventLogs(ctx context.Context, args *struct {
+	graphqlutil.ConnectionArgs
+}) (*userEventLogsConnectionResolver, error) {
+	// 🚨 SECURITY: Event logs can only be viewed by the user or site admin.
+	if err := backend.CheckSiteAdminOrSameUser(ctx, r.user.ID); err != nil {
+		return nil, err
+	}
+	var opt db.EventLogsListOptions
+	args.ConnectionArgs.Set(&opt.LimitOffset)
+	return &userEventLogsConnectionResolver{opt: opt, userID: r.user.ID}, nil
+}
+
+type userEventLogsConnectionResolver struct {
+	opt    db.EventLogsListOptions
+	userID int32
+}
+
+func (r *userEventLogsConnectionResolver) Nodes(ctx context.Context) ([]*userEventLogResolver, error) {
+	events, err := db.EventLogs.GetAllByUserID(ctx, r.userID)
+	if err != nil {
+		return nil, err
+	}
+
+	var eventLogs []*userEventLogResolver
```

Comment:
Since we already know the length of the slice, we could preallocate all of them.
```suggestion
	eventLogs := make([]*userEventLogResolver, 0, len(events))
```